### PR TITLE
Ccp 135

### PR DESCRIPTION
**Pull request by mkarthik90**
[CCP-135] Completed rdf generation through rest service.

### CODE REVIEW

File: Conceptpower+Spring/.classpath [hunk 1]
```diff
@@ -33,5 +33,6 @@
 			<attribute name="maven.pomderived" value="true"/>
 		</attributes>
 	</classpathentry>
+	<classpathentry kind="lib" path="/Users/karthikeyanmohan/Downloads/art-uniroma2-lime-api-062ce14de8b4.zip"/>
```
Comment (jdamerow) [inline Conceptpower+Spring/.classpath]: don't commit your classpath please

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/lucene/impl/LuceneUtility.java [hunk 2]
```diff
@@ -425,26 +425,24 @@ public void indexConcepts() throws LuceneException, IllegalArgumentException, Il
...
 
                 Document doc = reader.document(i);
-                String docId = doc.get("docId");
+                String docId = doc.get("word");
                 System.out.println(docId);
+
                 // do something with docId here...
             }
+        } catch (Exception ex) {
+            ex.printStackTrace();
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/lucene/impl/LuceneUtility.java]: handle exception

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptLookup.java [hunk 3]
```diff
@@ -81,4 +102,123 @@
...
+    public @ResponseBody ResponseEntity<String> getWordNetEntryInRdf(@PathVariable("word") String word,
+            @PathVariable("pos") String pos) {
+        ConceptEntry[] entries = null;
+        try {
+            entries = dictManager.getConceptListEntriesForWord(word, pos, null);
+        } catch (LuceneException ex) {
+            return new ResponseEntity<String>(ex.getMessage(), HttpStatus.BAD_REQUEST);
+        } catch (IllegalAccessException e) {
+            return new ResponseEntity<String>(e.getMessage(), HttpStatus.BAD_REQUEST);
+        }
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptLookup.java]: those two exceptions seem more like internal server errors than bad requests, or do they have something to do with request format/content?

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptLookup.java [hunk 4]
```diff
@@ -81,4 +102,123 @@
 
         return new ResponseEntity<String>(returnMsg.getXML(xmlEntries), HttpStatus.OK);
     }
+
+    @RequestMapping(value = "rest/ConceptLookupRdf/{word}/{pos}", method = RequestMethod.GET, produces = "application/xml")
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptLookup.java]: This should be available at the same endpoint as the regular xml (rest/ConceptLookup/{word}/{pos}). If you set produces to "application/rdf+xml", Spring will call this method when some goes to .ConceptLookup/word/pos.rdf.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptLookup.java [hunk 5]
```diff
@@ -81,4 +102,123 @@
...
+            return new ResponseEntity<String>(ex.getMessage(), HttpStatus.BAD_REQUEST);
+        } catch (IllegalAccessException e) {
+            return new ResponseEntity<String>(e.getMessage(), HttpStatus.BAD_REQUEST);
+        }
+        Model model = ModelFactory.createDefaultModel();
+        String syntax = "RDF/XML";
+        StringWriter out = new StringWriter();
+        for (ConceptEntry entry : entries) {
+            generateRDF(model, syntax, out, entry);
+        }
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptLookup.java]: the generation of RDF should be in its own class

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rdf/RDFMessageFactory.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
+package edu.asu.conceptpower.servlet.rdf;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import edu.asu.conceptpower.root.URIHelper;
+import edu.asu.conceptpower.servlet.xml.XMLTypeMessage;
+
+public class RDFMessageFactory {
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rdf/RDFMessageFactory.java]: where do you use this class and RDFConceptMessage?

**Comment by jdamerow** [comment]
also there seems to be a sysout somewhere that produces tons of output when calling the conceptlookup. please remove that.

**Comment by jdamerow** [comment]
please resolve conflicts

**Comment by mkarthik90** [comment]
closed this PR because the PR was created to CCP-128 instead of develop. Will create a new PR

**Pull request closed without merging**